core/vm/contracts.go

```diff
@@ -92,10 +90,8 @@ func (c *ecrecover) Run(in []byte) []byte {
 // SHA256 implemented as a native contract
 type sha256 struct{}
 
-func (c *sha256) RequiredGas(inputSize int) *big.Int {
-	n := big.NewInt(int64(inputSize+31) / 32)
-	n.Mul(n, params.Sha256WordGas)
-	return n.Add(n, params.Sha256Gas)
+func (c *sha256) RequiredGas(inputSize int) uint64 {
+	return uint64(inputSize+31)/32*params.Sha256WordGas + params.Sha256Gas
```

Comment:
Technically yes, but you'd need a significant amount of gas to make this overfow. There simply isn't enough ether to make it significant. Even if you'd pass it zeros only, you'd still need an enormous amount amount of ether to make the initial call.